### DISCUSSION TITLE
dimming window_rule

### DISCUSSION THREAD

**Discussion by name-snrl** [YaLTeR/niri#368] [Ideas]
Are there any plans to add this?

**Comment by YaLTeR** [comment]
What do you mean by dimming? Darkening?

**Comment by name-snrl** [comment]
> Darkening?

Yes, at current time I use swayfx with [`default_dim_inactive`](https://github.com/WillPower3309/swayfx/blob/3c621dec7d653231f960d377fcb3ceeed55953e2/sway/sway.5.scd?plain=1#L660) option. I first encountered this on hyprland ([`dim_inactive`](https://wiki.hyprland.org/Configuring/Variables/#decoration)). Unlike transparency, this only shades windows, which keeps the window content readable. I prefer to use this instead of borders to detect a focused window

**Reply by YaLTeR** [reply]
Thanks. We'll have to see. Because this needs a custom shader, at which point maybe it's better to just go all the way and expose custom shaders for windows. That is however considerably more complex than just dimming, so maybe it's better to add dimming separately.

**Reply by lpnh** [reply]
FWIW, perhaps [dim](https://github.com/marcelohdez/dim) can provide some inspiration. Note that it relies on the Single-pixel buffer protocol, i.e. the `wp_single_pixel_buffer_manager_v1` interface.
